### ISSUE TITLE
Efficient batched 2D inference by pooling tiles across images

### ISSUE THREAD

**Issue by gwcmowry** [MouseLand/cellpose#1460]
## Before you fill out this form:
**Did you review the FAQ?**
- [x] Yes, I reviewed the FAQ on the [cellpose ReadTheDocs](https://cellpose.readthedocs.io/en/latest/faq.html) 

**Did you look through previous (open AND closed) issues posted on GH?**
- [x] Yes, I [searched for related previous GH issues](https://github.com/MouseLand/cellpose/issues?q=is%3Aissue).

### **Is your feature request related to a problem? Please describe.**
A common use case in my lab is segmenting many 2D images from a folder, which is currently quite slow. Existing sequential segmentation of 2D images does not saturate the GPU well nor does it amortize launch kernel costs.

### **Describe the solution you'd like**
Pooling tile across multiple images for efficient batch inference for multiple 2D images. Stacking multiple 2D images into a single 3D one for inference does not work for this, since 3D segmentation is not intended.

### **Describe alternatives you've considered**
I considered using `model.eval(list_of_images)`, but in Cellpose v4 this appears to loop over images and call `eval()` recursively, so it does not substantially amortize CellposeSAM forward-pass overhead across a folder.

I also tried stacking multiple 2D images into a single array, but that goes through Cellpose’s stack/3D-style mask handling and produces confusing “3D stack” behavior/logging even when `do_3D=False`. I wanted to preserve strictly independent 2D mask computation.

I also tested changing segmentation parameters such as diameter, flow threshold, and smoothing. Those can improve speed, but they may change masks. The proposed approach is intended to preserve equivalent masks by changing only how network tiles are batched.

### **Additional context**
I am happy to contribute a pull request for this, I already have a prototype showing a roughly 1.6x speedup on mps over sequential per-image segmentation.

**Comment by mrariden** [comment]
Cellpose is capable of running a stack of images with the `do_3D=False` argument if you pass a numpy array: 

```python
img = cellpose.io.imread('an/image/file.png')
np_img = np.array([img] * 10)
cpm = models.CellposeModel(gpu=True)
masks, _, _ = cpm.eval(np_img, do_3D=False)

for mask in masks: 
    plt.imshow(mask)
    plt.show()
```

You'll get a warning saying that `norm3D` is set to `True` (which is a default) but the data isn't 3D, so cellpose will set it to `False`. You can remove that warning by explicitly passing `normalize={'norm3D':False}` to `.eval()`

I'd be happy to review any performance improvements.

**Comment by carsen-stringer** [comment]
Going to close this due to inactivity but please feel free to comment @gwcmowry if you have any suggestions
